pkg/ztunnel/iptables/inpod.go

```diff
@@ -0,0 +1,480 @@
+}
+
+func (m *ruleManager) addVersioned(ipv4, ipv6, table, chain string, parameters ...string) {
+	m.rules = append(m.rules, rule{ipv4, ipv6, table, chain, parameters})
+}
+
+func replaceIPPlaceholder(args []string, ip string) []string {
+	if ip == "" {
+		return args
+	}
+	res := make([]string, len(args))
+	copy(res, args)
+	for i, arg := range args {
+		if arg == VersionSpecificPlaceholder {
+			res[i] = ip
+		}
+	}
+	return res
+}
+
+func (m *ruleManager) install(ipv4Enabled, ipv6Enabled bool) error {
+	for _, rule := range m.rules {
+		if ipv4Enabled {
+			ruleSpec := replaceIPPlaceholder(rule.parameters, rule.ipv4)
+			// Check if rule exists
+			exists, err := m.ipt4.Exists(rule.table, rule.chain, ruleSpec...)
+			if err != nil {
+				return fmt.Errorf("failed to check iptables rule existence: %w", err)
+			}
+			if !exists {
+				// Rule doesn't exist, add it
+				if err := m.ipt4.Append(rule.table, rule.chain, ruleSpec...); err != nil {
+					return fmt.Errorf("failed to insert iptables rule (%s %s %v): %w", rule.table, rule.chain, ruleSpec, err)
+				}
+			}
+		}
+
+		if ipv6Enabled && m.ipt6 != nil {
+			ruleSpec := replaceIPPlaceholder(rule.parameters, rule.ipv6)
+			// Check if rule exists
+			exists, err := m.ipt6.Exists(rule.table, rule.chain, ruleSpec...)
+			if err != nil {
+				return fmt.Errorf("failed to check ip6tables rule existence: %w", err)
+			}
+			if !exists {
+				// Rule doesn't exist, add it
+				if err := m.ipt6.Append(rule.table, rule.chain, ruleSpec...); err != nil {
+					return fmt.Errorf("failed to insert ip6tables rule (%s %s %v): %w", rule.table, rule.chain, ruleSpec, err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (m *ruleManager) createChains(ipv4Enabled, ipv6Enabled bool) error {
+	for _, table := range []string{"mangle", "nat"} {
+		for _, chain := range []string{InpodPreroutingChain, InpodOutputChain} {
+			if ipv4Enabled {
+				// Check if chain exists first
+				chainExists, err := m.ipt4.ChainExists(table, chain)
+				if err != nil {
+					return fmt.Errorf("failed to check iptables chain existence: %w", err)
+				}
+
+				if !chainExists {
+					// Chain doesn't exist, create it
+					if err := m.ipt4.NewChain(table, chain); err != nil {
+						return fmt.Errorf("failed to create iptables chain %s: %w", chain, err)
+					}
+				}
+			}
+
+			if ipv6Enabled && m.ipt6 != nil {
+				// Check if chain exists first
+				chainExists, err := m.ipt6.ChainExists(table, chain)
+				if err != nil {
+					return fmt.Errorf("failed to check ip6tables chain existence: %w", err)
+				}
+
+				if !chainExists {
+					// Chain doesn't exist, create it
+					if err := m.ipt6.NewChain(table, chain); err != nil {
+						return fmt.Errorf("failed to create ip6tables chain %s: %w", chain, err)
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func addInPodRules(logger *slog.Logger, ipv4Enabled, ipv6Enabled bool) error {
+	var ipt4, ipt6 *iptables.IPTables
```

Comment:
nit: I would personally find it help if instead of sprinkling comments about the individual rule creations through the function, you tell the reader what the rules we are creating, what they do, and any conditional aspects of rule creation as one comment either at this function or somewhere in the package. 

I find having a summary of what even 'in pod' rules are and what they are doing would help both reviewers and future editors of the code.